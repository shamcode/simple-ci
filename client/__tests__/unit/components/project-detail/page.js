import { DI } from 'sham-ui';
import Page from '../../../../src/components/project-detail/page';
import * as directives from 'sham-ui-directives';
import hrefto from 'sham-ui-router/href-to';
import renderer from 'sham-ui-test-helpers';

afterEach( () => {
    DI.bind( 'store', null );
    DI.bind( 'router', null );
} );

it( 'renders correctly', () => {
    const storeMock = jest.fn().mockReturnValue( Promise.resolve( { project: { id: 1 } } ) );
    DI.bind( 'store', {
        getProjectById: storeMock
    } );
    DI.bind( 'router', {
        storage: { params: { id: 1 } },
        generate: jest.fn().mockReturnValueOnce( '/' )
    } );

    const meta = renderer( Page, {
        directives: {
            ...directives,
            hrefto
        }
    } );
    expect( storeMock.mock.calls[ 0 ][ 0 ] ).toBe( 1 );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'render errors', () => {
    const storeMock = jest.fn().mockReturnValue( Promise.reject() );
    DI.bind( 'store', {
        getProjectById: storeMock
    } );
    DI.bind( 'router', {
        storage: { params: {} },
        generate: jest.fn().mockReturnValueOnce( '/' )
    } );

    const meta = renderer( Page, {
        directives: {
            ...directives,
            hrefto
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

