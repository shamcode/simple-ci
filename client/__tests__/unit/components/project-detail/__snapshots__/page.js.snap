// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render errors 1`] = `
Object {
  "Constructor": "ProjectDetailPage",
  "Options": Object {
    "ID": "component",
    "dataLoaded": false,
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "errors": Array [],
    "project": Object {},
  },
  "html": "
 <ul class=\\"breadcrumb\\">
   <li>
     <a href=\\"/\\"> Projects </a>
   </li>
   <!--if-->
   <li></li>
 </ul>
 <!--BreadCrumb-->
 <!--breadcrumbs-->
 <!--if-->
 <!--WithoutFlushLoader-->
 <!--if-->
 <!--PageContent-->
",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Constructor": "ProjectDetailPage",
  "Options": Object {
    "ID": "component",
    "dataLoaded": false,
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "errors": Array [],
    "project": Object {},
  },
  "html": "
 <ul class=\\"breadcrumb\\">
   <li>
     <a href=\\"/\\"> Projects </a>
   </li>
   <!--if-->
   <li></li>
 </ul>
 <!--BreadCrumb-->
 <!--breadcrumbs-->
 <!--if-->
 <!--WithoutFlushLoader-->
 <!--if-->
 <!--PageContent-->
",
}
`;
